src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+If you need to review the steps for creating a new IntelliJ project, refer
+back to the :ref:`First Java Program <create-new-java-project>` instructions.
+
+#. Create a new project and call it ``TempConverter``.
+#. Right-click (or control-click) on the ``src`` folder. Select *New*, then
+   *Java Class*.
+
+   .. figure:: figures/create-new-class.png
+      :alt: Create the TempConverter class.
+
+#. You should see the following empty class in the editor panel:
+
+   .. figure:: figures/TempConverter-class.png
+      :alt: New TempConverter class.
+
+#. Cut and paste the following code into the editor. We will analyze the
+   different statements in a moment.
+
+   .. sourcecode:: java
+      :linenos:
+
+      import java.util.Scanner;
+
+      public class TempConverter {
+         public static void main(String[] args) {
+            double fahrenheit;
+            double celsius;
+            Scanner input;
+
+            input = new Scanner(System.in);
+            System.out.println("Enter the temperature in Fahrenheit: ");
+            fahrenheit = input.nextDouble();
+            input.close();
+
+            celsius = (fahrenheit - 32) * 5/9;
+            System.out.println("The temperature in Celsius is: " + celsius + "°C");
+         }
+      }
+
+#. Build and run the program to verify that it works. Entering a Fahrenheit
+   temperature of ``212`` yields the result, ``The temperature in Celsius is:
+   100.0°C``.
+
+There are several new concepts introduced in this example. We will look
+at them in the following order:
+
+#. The ``import`` statement
+#. Declaring variables
+#. Collecting input with the Scanner class
+
+``import``
+-----------
+
+The ``import`` statement in Java allows us to access the class, methods, and
+data stored a different file. ``import`` tells the compiler that we are going
+to use a shortened version of the class name. In this example, we are going to
+use the class ``java.util.Scanner``, but we can refer to it as just
+``Scanner``. We could use the ``java.util.Scanner`` class without any problem
+and without any import statement, provided that we always referred to it by its
+full name.
+
+In Java, you can use any class that is available WITHOUT having to import the
+class, but you must adhere to two very important conditions:
+
+#. The ``javac`` and ``java`` commands must know that the class exists.
+#. You must use the full name of the class.
+
+.. admonition:: Try It
+
+   In the sample code, remove the ``import`` statement in line 1, and change
+   ``Scanner`` on lines 7 & 9 to ``java.util.Scanner``. The program should
+   still compile and run.
+
+The class naming system in Java is very hierarchical. The *full* name of the
+``Scanner`` class is really ``java.util.Scanner``. You can think of this name
+as having two parts:
+
+#. ``java.util`` is called the *package*,
+#. ``Scanner`` is the class name.
+
+We’ll talk more about the class naming system a bit later.
+
+Java Knows Which Classes Exist, Kinda
```

Comment:
im not crazy about stating that java knows anything. its not sentient. are you open to changing this language to address what classes can be used by a java program in a given setting or environment?

Reply:
I was just trying to add a little subtle levity to the content, but we can avoid Skynet-type personification.

Suggestions?